New package ArobasMusic.GuitarPro.7.Soundbanks version 1.1.123

marco-eckstein opened the pull request:
###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/winget-pkgs/pull/132558)

marco-eckstein committed:
New package ArobasMusic.GuitarPro.7.Soundbanks version 1.1.123

See https://github.com/microsoft/winget-pkgs/discussions/132064.

Trenly commented on manifests/a/ArobasMusic/GuitarPro/7/Soundbanks/1.1.123/ArobasMusic.GuitarPro.7.installer.yaml:
```diff
@@ -0,0 +1,23 @@
+UpgradeBehavior: install
+Dependencies:
+  PackageDependencies:
+    - PackageIdentifier: ArobasMusic.GuitarPro.7
```
Won't this install the same package twice, since the InstallerURL of ArobasMusic.GuitarPro.7 is the same as this one?

marco-eckstein replied:
Probably. I'll remove that.

marco-eckstein committed:
Update package ArobasMusic.GuitarPro.7.Soundbanks version 1.1.123: remove dependencies

Addresses https://github.com/microsoft/winget-pkgs/pull/132558#discussion_r1440720416.

marco-eckstein commented:
@trenly: The manifest is valid. I guess the `PackageIdentifier` and folder structure are the problem here? What would you suggest?

stephengillie commented:
Automatic Validation ended with:
>  MSI (s) (3C:04) [10:17:21:447]: Executing op: ComponentRegister(ComponentId={D29D9058-F6B3-5A55-8CDD-74F5727D9617},KeyPath=22*85*\SOFTWARE\Microsoft\Windows\Windows Error Reporting\RuntimeExceptionHelperModules\C:\Program Files\dotnet\shared\Microsoft.NETCore.App\6.0.25\mscordaccore.dll,State=3,,Disk=1,SharedDllRefCount=0,BinaryType=1) MSI (s) (3C:04) [10:17:23:197]: Executing op: RegOpenKey(Root=-2147483646,Key=SOFTWARE\Microsoft\Windows\Windows Error Reporting\RuntimeExceptionHelperModules,,BinaryType=1,,) Property(S): ErrorDialog = ErrorDlg MSI (s) (3C:04) [10:17:29:478]: Windows Installer installed the product. Product Name: Microsoft .NET Runtime - 6.0.25 (x64). Product Version: 48.100.4028. Product Language: 1033. Manufacturer: Microsoft Corporation. Installation success or error status: 0. MSI (s) (3C:44) [10:17:38:306]: Windows Installer installed the product. Product Name: Microsoft .NET Host FX Resolver - 6.0.25 (x64). Product Version: 48.100.4028. Product Language: 1033. Manufacturer: Microsoft Corporation. Installation success or error status: 0. MSI (s) (3C:EC) [10:17:41:697]: Windows Installer installed the product. Product Name: Microsoft .NET Host - 6.0.25 (x64). Product Version: 48.100.4028. Product Language: 1033. Manufacturer: Microsoft Corporation. Installation success or error status: 0. MSI (s) (3C:B8) [10:17:47:088]: Windows Installer installed the product. Product Name: Microsoft Windows Desktop Runtime - 6.0.25 (x64). Product Version: 48.100.4037. Product Language: 1033. Manufacturer: Microsoft Corporation. Installation success or error status: 0. MSI (s) (3C:C8) [10:19:21:494]: Windows Installer installed the product. Product Name: Microsoft Visual C++ 2022 X64 Minimum Runtime - 14.38.33130. Product Version: 14.38.33130. Product Language: 1033. Manufacturer: Microsoft Corporation. Installation success or error status: 0. MSI (s) (3C:74) [10:19:22:525]: Windows Installer installed the product. Product Name: Microsoft Visual C++ 2022 X64 Additional Runtime - 14.38.33130. Product Version: 14.38.33130. Product Language: 1033. Manufacturer: Microsoft Corporation. Installation success or error status: 0. [16E4:0F24][2024-01-03T10:19:19]w363: Could not create system restore point, error: 0x80070422. Continuing... Action start 10:19:43: AI_ProcessFailActions. Action start 10:19:43: AI_ConfigFailActions. Action ended 10:19:43: AI_ConfigFailActions. Return value 1. Action ended 10:19:43: AI_ProcessFailActions. Return value 1. MSI (s) (2C:68) [10:16:33:140]: Executing op: ComponentRegister(ComponentId={08CDD9F7-522C-5B95-9381-DDAB637EF797},KeyPath=02*85*\SOFTWARE\Microsoft\Windows\Windows Error Reporting\RuntimeExceptionHelperModules\C:\Program Files\dotnet\shared\Microsoft.NETCore.App\6.0.25\mscordaccore.dll,State=3,,Disk=1,SharedDllRefCount=0,BinaryType=0) MSI (s) (2C:68) [10:16:34:906]: Executing op: RegOpenKey(Root=-2147483646,Key=SOFTWARE\Microsoft\Windows\Windows Error Reporting\RuntimeExceptionHelperModules,,BinaryType=0,,) MSI (s) (2C:68) [10:16:35:156]: Windows Installer installed the product. Product Name: Microsoft .NET Runtime - 6.0.25 (x86). Product Version: 48.100.4028. Product Language: 1033. Manufacturer: Microsoft Corporation. Installation success or error status: 0. MSI (s) (2C:6C) [10:16:35:749]: Windows Installer installed the product. Product Name: Microsoft .NET Host FX Resolver - 6.0.25 (x86). Product Version: 48.100.4028. Product Language: 1033. Manufacturer: Microsoft Corporation. Installation success or error status: 0. MSI (s) (2C:64) [10:16:36:437]: Windows Installer installed the product. Product Name: Microsoft .NET Host - 6.0.25 (x86). Product Version: 48.100.4028. Product Language: 1033. Manufacturer: Microsoft Corporation. Installation success or error status: 0. MSI (s) (2C:68) [10:16:40:640]: Windows Installer installed the product. Product Name: Microsoft Windows Desktop Runtime - 6.0.25 (x86). Product Version: 48.100.4037. Product Language: 1033. Manufacturer: Microsoft Corporation. Installation success or error status: 0. MSI (s) (2C:A8) [10:17:53:874]: Windows Installer installed the product. Product Name: Microsoft Visual C++ 2022 X86 Minimum Runtime - 14.38.33130. Product Version: 14.38.33130. Product Language: 1033. Manufacturer: Microsoft Corporation. Installation success or error status: 0. MSI (s) (2C:4C) [10:17:54:671]: Windows Installer installed the product. Product Name: Microsoft Visual C++ 2022 X86 Additional Runtime - 14.38.33130. Product Version: 14.38.33130. Product Language: 1033. Manufacturer: Microsoft Corporation. Installation success or error status: 0. [1B78:0E6C][2024-01-03T10:17:52]w363: Could not create system restore point, error: 0x80070422. Continuing... Action start 10:18:18: AI_ProcessFailActions. Action start 10:18:18: AI_ConfigFailActions. Action ended 10:18:18: AI_ConfigFailActions. Return value 1. Action ended 10:18:18: AI_ProcessFailActions. Return value 1.

(Automated response - build 516.)

stephengillie commented:
![image](https://github.com/microsoft/winget-pkgs/assets/114952565/224aedf0-6ba3-4512-9e62-e65af704b357)

stephengillie commented on manifests/a/ArobasMusic/GuitarPro/7/Soundbanks/1.1.123/ArobasMusic.GuitarPro.7.installer.yaml:
```diff
@@ -0,0 +1,20 @@
+  - silentWithProgress
+UpgradeBehavior: install
+Installers:
+  - Architecture: x86
```
```suggestion
  - Architecture: x64
```
Since `x86` validations run on 32-bit OSes, I'm guessing this might be a 64-bit application.

marco-eckstein replied:
[The publisher says](https://support.guitar-pro.com/hc/en-us/articles/360001599877-GP7-6-Download-Guitar-Pro-7-6-Tablature-Editor-Software) _Architecture 32 or 64 bits_. The main software (that uses the same installer) `ArobasMusic.GuitarPro.7` has been accepted with `Architecture: x86`. So I think it must be `neutral` or `x86`.

stephengillie commented:
>  Manifest: The filename must match the PackageIdentifier and ManifestType properties from the manifest. actual 'ArobasMusic.GuitarPro.7.yaml', expected 'ArobasMusic.GuitarPro.7.Soundbanks.yaml'

stephengillie commented on manifests/a/ArobasMusic/GuitarPro/7/Soundbanks/1.1.123/ArobasMusic.GuitarPro.7.installer.yaml:
```diff
@@ -0,0 +1,20 @@
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.installer.1.5.0.schema.json
+
+PackageIdentifier: ArobasMusic.GuitarPro.7.Soundbanks
```
```suggestion
PackageIdentifier: ArobasMusic.GuitarPro.7
```

stephengillie commented on manifests/a/ArobasMusic/GuitarPro/7/Soundbanks/1.1.123/ArobasMusic.GuitarPro.7.locale.en-US.yaml:
```diff
@@ -0,0 +1,20 @@
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.defaultLocale.1.5.0.schema.json
+
+PackageIdentifier: ArobasMusic.GuitarPro.7.Soundbanks
```
```suggestion
PackageIdentifier: ArobasMusic.GuitarPro.7
```

stephengillie commented on manifests/a/ArobasMusic/GuitarPro/7/Soundbanks/1.1.123/ArobasMusic.GuitarPro.7.yaml:
```diff
@@ -0,0 +1,7 @@
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.version.1.5.0.schema.json
+
+PackageIdentifier: ArobasMusic.GuitarPro.7.Soundbanks
```
```suggestion
PackageIdentifier: ArobasMusic.GuitarPro.7
```

marco-eckstein commented:
> ![image](https://private-user-images.githubusercontent.com/114952565/294313902-224aedf0-6ba3-4512-9e62-e65af704b357.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQ0MDgwOTIsIm5iZiI6MTcwNDQwNzc5MiwicGF0aCI6Ii8xMTQ5NTI1NjUvMjk0MzEzOTAyLTIyNGFlZGYwLTZiYTMtNDUxMi05ZTYyLWU2NWFmNzA0YjM1Ny5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMTA0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDEwNFQyMjM2MzJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03MGIyNDQzZWI2NzliYzFjYzNkYmNmM2YzYWY1NDg1MjZlZjkzOTAwOWYyYzM5NWJkMGJlYzE4NDBlMjE0OGQ1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.YL8atVduz7LAYrYfdu3QIb5l-LW8MSyoL2E5x1RejY0)

The message says _Guitar Pro 8_. This is a different package. I don't see how it is related to this package.

marco-eckstein commented:
> > Manifest: The filename must match the PackageIdentifier and ManifestType properties from the manifest. actual 'ArobasMusic.GuitarPro.7.yaml', expected 'ArobasMusic.GuitarPro.7.Soundbanks.yaml'

`ArobasMusic.GuitarPro.7` is a different package which has the same installer. Please see [the discussion](https://github.com/microsoft/winget-pkgs/discussions/132064) I had mentioned in my commit message. Therein, @Trenly had suggested to create this separate package. Recap: The installer adds two entries to Apps & Features, both of which can/must be uninstalled separately.

stephengillie commented:
> > ![image](https://private-user-images.githubusercontent.com/114952565/294313902-224aedf0-6ba3-4512-9e62-e65af704b357.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQ0MDgwOTIsIm5iZiI6MTcwNDQwNzc5MiwicGF0aCI6Ii8xMTQ5NTI1NjUvMjk0MzEzOTAyLTIyNGFlZGYwLTZiYTMtNDUxMi05ZTYyLWU2NWFmNzA0YjM1Ny5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMTA0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDEwNFQyMjM2MzJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03MGIyNDQzZWI2NzliYzFjYzNkYmNmM2YzYWY1NDg1MjZlZjkzOTAwOWYyYzM5NWJkMGJlYzE4NDBlMjE0OGQ1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.YL8atVduz7LAYrYfdu3QIb5l-LW8MSyoL2E5x1RejY0)
> 
> The message says _Guitar Pro 8_. This is a different package. I don't see how it is related to this package.

I had the wrong PR, sorry.

stephengillie commented:
> > > Manifest: The filename must match the PackageIdentifier and ManifestType properties from the manifest. actual 'ArobasMusic.GuitarPro.7.yaml', expected 'ArobasMusic.GuitarPro.7.Soundbanks.yaml'
> 
> `ArobasMusic.GuitarPro.7` is a different package which has the same installer. Please see [the discussion](https://github.com/microsoft/winget-pkgs/discussions/132064) I had mentioned in my commit message. Therein, @Trenly had suggested to create this separate package. Recap: The installer adds two entries to Apps & Features, both of which can/must be uninstalled separately.

To clear the `Manifest-Validation-Error` label, add the `.Soundbanks` part of the PackageIdentifier to the file path. The two have to match, but currently they don't.

marco-eckstein commented:
> To clear the `Manifest-Validation-Error` label, add the `.Soundbanks` part of the PackageIdentifier to the file path. The two have to match, but currently they don't.

I don't get it - where is the mismatch?
```
ArobasMusic.GuitarPro.7.Soundbanks
ArobasMusic\GuitarPro\7\Soundbanks
```

marco-eckstein committed:
Update package ArobasMusic.GuitarPro.7.Soundbanks version 1.1.123: fix file names

marco-eckstein commented:
> To clear the `Manifest-Validation-Error` label, add the `.Soundbanks` part of the PackageIdentifier to the file path. The two have to match, but currently they don't.

Ah, I see, it's the file names. I didn't see the forest for the trees. It's fixed now.

marco-eckstein commented:
> Possible duplicate package entry.
> 
> Similar installer SHA256 hash found in manifest:
> 
>     * `InstallerSha256: c3e508e53760633fa1cf17bdeb30d037bd7ea16c79bce3892cd33c74740c61c6`
>       
>       * manifests\a\ArobasMusic\GuitarPro\7\7.6.0.2089

This is to be expected. It is OK. See comments/discussion.

stephengillie commented:
I don't see any reason for the [Policy 1.8](https://learn.microsoft.com/en-us/windows/package-manager/package/windows-package-manager-policies#18-financial-transactions) check.

stephengillie commented:
@wingetbot waivers Add Policy-Test-1.8

stephengillie commented:
/AzurePipelines run
